### PR TITLE
chore: setting up flamegraph

### PR DESCRIPTION
**Pull request by benfdking**

### CODE REVIEW

**Comment by gvozdvmozgu** [comment]
It seems that the pprof crate only supports Unix-like systems, as it fails to compile on Windows.

File: crates/lib/Cargo.toml [hunk 1]
```diff
@@ -57,3 +57,4 @@ expect-test = "1.5"
 glob = "0.3"
 serde_json = "1"
 tempdir = "0.3"
+pprof = { version = "0.13", features = ["flamegraph", "criterion"] }
```
Comment (nikolaevx) [inline crates/lib/Cargo.toml:61]:
```suggestion
[target.'cfg(unix)'.dependencies]
pprof = { version = "0.13", features = ["flamegraph", "criterion"] }
```

File: crates/lib/benches/depth_map.rs [hunk 2]
```diff
@@ -1,4 +1,5 @@
 use criterion::{black_box, criterion_group, criterion_main, Criterion};
+use pprof::criterion::{Output, PProfProfiler};
```
Comment (nikolaevx) [inline crates/lib/benches/depth_map.rs:3]:
```suggestion
#[cfg(unix)]
use pprof::criterion::{Output, PProfProfiler};
```

File: crates/lib/benches/depth_map.rs [hunk 3]
```diff
@@ -75,5 +76,9 @@ fn depth_map(c: &mut Criterion) {
     });
 }
 
-criterion_group!(benches, depth_map);
+criterion_group! {
+    name = benches;
+    config = Criterion::default().with_profiler(PProfProfiler::new(100, Output::Flamegraph(None)));
+    targets = depth_map
+}
```
Comment (nikolaevx) [inline crates/lib/benches/depth_map.rs:85]:
```suggestion
#[cfg(unix)]
criterion_group! {
    name = benches;
    config = Criterion::default().with_profiler(PProfProfiler::new(100, Output::Flamegraph(None)));
    targets = depth_map
}
```

**Review by gvozdvmozgu — APPROVED** [review]

**Pull request merged**